build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/ProcessMSBuildTiming.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class ProcessMSBuildTiming : Task
+	{
+		[Required]
+		public ITaskItem[] InputFiles { get; set; }
+
+		public string ResultsFilename { get; set; }
+
+		public bool AddResults { get; set; }
+
+		public string LabelSuffix { get; set; }
+
+		Dictionary<string, string> results = new Dictionary<string, string> ();
+
+		public override bool Execute ()
+		{
+			foreach (var file in InputFiles) {
+				using (var stream = File.OpenRead (file.ItemSpec)) {
+					var element = XElement.Load (stream);
```

Comment:
Why not use [`XElement.Load(string)`](https://msdn.microsoft.com/en-us/library/bb298435(v=vs.110).aspx)? This would remove a level of indentation for the `using` block.